packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmObjectHelper.kt

```diff
@@ -44,6 +46,16 @@ object RealmObjectHelper {
         return RealmInterop.realm_get_value(o, key)
     }
 
+    @Suppress("unused") // Called from generated code
+    internal fun <R> getTimestamp(obj: RealmObjectInternal, col: String): RealmInstant? {
+        obj.checkValid()
+        val realm = obj.`$realm$Owner` ?: throw IllegalStateException("Invalid/deleted object")
+        val o = obj.`$realm$ObjectPointer` ?: throw IllegalStateException("Invalid/deleted object")
+        val key = RealmInterop.realm_get_col_key(realm.dbPointer, obj.`$realm$TableName`!!, col)
+        val res = RealmInterop.realm_get_value<Timestamp>(o, key)
+        return if (res == null) null else RealmInstantImpl(res)
```

Comment:
We could consider utilizing typemaps here to be able to generalize the path. But for speed I guess it would be better specializing the path. 

When refactored for the cached key based look up I think it would be relatively clean to just have a specialized variant for each type and by inlining it we skip the tedious modifications of the compiler plugin. Unless there would be a benefit in keeping the logic in library implementation instead of as compile time generated parts of user models 🤔

Reply:
Yes, I think there is room for improvements here, but we should probably do this as part of cleaning this up in general. So would probably just prefer to keep this here for now.